tools/autotools/tsk_logical_imager.cpp

```diff
@@ -734,6 +767,41 @@ main(int argc, char **argv1)
     }
 
     ruleSet = new LogicalImagerRuleSet(toNarrow(configFilename));
+
+    TskHelper::getInstance().setImgInfo(img);
+    TSK_VS_INFO *vs_info;
+    if ((vs_info = tsk_vs_open(img, 0, TSK_VS_TYPE_DETECT)) == NULL) {
+        std::cout << "No volume system found. Looking for file system" << std::endl;
+        openFs(img, 0);
+    } else {
+        // process the volume system
+        for (TSK_PNUM_T i = 0; i < vs_info->part_count; i++) {
+            const TSK_VS_PART_INFO *vs_part = tsk_vs_part_get(vs_info, i);
+            std::cout << "Partition: " + string(vs_part->desc) + "    Start: " + std::to_string(vs_part->start) << std::endl;
+            if ((vs_part->flags & TSK_VS_PART_FLAG_UNALLOC) || (vs_part->flags & TSK_VS_PART_FLAG_META)) {
+                continue;
+            }
+            openFs(img, vs_part->start * vs_part->vs->block_size);
+        }
+        tsk_vs_close(vs_info);
+    }
+
+    const std::list<TSK_FS_INFO *> fsList = TskHelper::getInstance().getFSInfoList();
+    TSKFileNameInfo filenameInfo;
+    const std::vector<std::string> filePaths = ruleSet->getFilePaths();
+    TSK_FS_FILE *fs_file;
+    for (std::list<TSK_FS_INFO *>::const_iterator fsListIter = fsList.begin(); fsListIter != fsList.end(); ++fsListIter) {
+        for (std::vector<std::string>::const_iterator iter = filePaths.begin(); iter != filePaths.end(); ++iter) {
+            int retval = TskHelper::getInstance().path2Inum(*fsListIter, iter->c_str(), filenameInfo, NULL, &fs_file);
+            fprintf(stdout, "Path2Inum returns %d %s for %s\n", retval, (retval == 0 && fs_file == NULL ? "duplicate" : ""), iter->c_str());
+            if (retval == 0 && fs_file != NULL) {
+                (void) TskFindFiles::extractFile(fs_file);
+                delete fs_file;
+            }
+        }
+    }
+    TskHelper::getInstance().reset();
```

Comment:
Perhaps also call reset at the beginning, just to make sure there's nothing in there?